pkg/telemetry/cassandra_metrics_filters_test.go

```diff
@@ -83,14 +85,15 @@ func Test_InjectCassandraTelemetryFilters_Disabled(t *testing.T) {
 	}
 
 	// Test with a nil TelemetrySpec
-	var telemetrySpec *telemetry.TelemetrySpec
+	telemetrySpec := &telemetry.TelemetrySpec{
+		Mcac: &telemetry.McacTelemetrySpec{
+			MetricFilters: &[]string{},
+		},
+	}
 	InjectCassandraTelemetryFilters(telemetrySpec, dcConfig)
 	cassandraEnvVariables := dcConfig.PodTemplateSpec.Spec.Containers[0].Env
-	assert.Equal(t, 0, len(cassandraEnvVariables), "Expected no env variable to be injected")
+	assert.Equal(t, 1, len(cassandraEnvVariables), "Expected 1 env variable to be injected")
+	assert.Equal(t, cassandraEnvVariables[0].Name, "METRIC_FILTERS", "Expected METRIC_FILTERS env variable to be injected")
```

Comment:
Nit: This looks brittle. If the ordering of the environment variable injection changes you'll get a false positive here. Didn't you have a findEnvironmentVariable function in here somewhere that you could use to make this more robust?

Reply:
In this very case we should be fine since it's a unit test and no controller is involved.
In the e2e tests, I'm using a function to find the environment variable I'm looking for, because its position is unpredictable indeed.